README.md

```diff
@@ -114,6 +114,21 @@ AdaScale can be used to wrap a SGD optimizer and to be used in DDP (Distributed
 training or non-DDP with gradient accumulation. The benefit is to re-use the same LR
 schedule from a baseline batch size when effective batch size is bigger.
 
+```python
+from torch.optim import SGD
+from fairscale.optim import AdaScale
+
+...
+optim = AdaScale(SGD(model.parameters(), lr=0.1))
+...
+step = 0
+for sample in dataloader:
+    ...
+    step += optim.gain()
+    update_lr(step)  # This update_lr() function is unchanged from single GPU training.
```

Comment:
The standard PyTorch learning rate schedulers (in `torch.optim.lr_scheduler`) have a different syntax. Instead of passing `step` as an argument, they use a syntax like:
```
>>> optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
>>> scheduler = torch.optim.lr_scheduler.CyclicLR(optimizer, base_lr=0.01, max_lr=0.1)
>>> data_loader = torch.utils.data.DataLoader(...)
>>> for epoch in range(10):
>>>     for batch in data_loader:
>>>         train_batch(...)
>>>         scheduler.step()
```
I have an idea of how we could fix this pretty easily. We should be keeping track of the scale-invariant step number internally to AdaScale (accumulate the gain every time `step()`) is called. Then we could add a method to AdaScale that returns the number of times to call `scheduler.step()` after calling `optim.step()`.

One other point - the example should probably include the call to `optim.step()`

Reply:
I see. That's a great observation! I didn't know about the standard way either. I will give this a try in the example. Stay tuned for new version of this PR later.